aten/src/ATen/native/cuda/Blas.cpp

```diff
@@ -819,9 +819,97 @@ static bool _scaled_mm_allowed_device() {
 #endif
 }
 
-// Computes matrix multiply + bias while applying scaling to input and output matrices and computes amax
+namespace{
+
+enum class ScalingType {
+  TensorWise,
+  RowWise,
+  Error
+};
+/*
+ * Scaling Type Determination:
+ * ---------------------------
+ * Conditions and corresponding Scaling Types:
+ *
+ * - If scale_a.numel() == 1 && scale_b.numel() == 1:
+ *   - Returns TensorWise.
+ *
+ * - Else if scale_a.dim() == 1 && scale_a.size(0) == dim_m && scale_b.size(0) == dim_n:
```

Comment:
Perhaps also add a comment for the restriction of Tensor's being required to be 2D for the RowWise case